docs/upgrade/read-model-for-product-lists.md

```diff
@@ -0,0 +1,99 @@
+    ```
+    - Note: the bundle needs to be registered after `ShopsysFrameworkBundle` as it overwrites its `image` Twig function
+- in all frontend controllers where you are listing products, use the new `Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFacadeInterface`:
+    - in `ProductController::detailAction`:
+        ```diff
+        - $accessories = $this->productOnCurrentDomainFacade->getAccessoriesForProduct($product);
+        + $accessories = $this->listedProductsFacade->getAllAccessories($product->getId());
+        ```
+    - in `ProductController::listByCategoryAction`:
+        ```diff
+        - $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsInCategory(
+        -      $productFilterData,
+        -      $orderingModeId,
+        -      $page,
+        -      self::PRODUCTS_PER_PAGE,
+        -      $id
+        -  );
+        + $paginationResult = $this->listedProductsFacade->getFilteredPaginatedInCategory(
+        +      $id,
+        +      $productFilterData,
+        +      $orderingModeId,
+        +      $page,
+        +      self::PRODUCTS_PER_PAGE
+        +  );
+        ```
+    - in `ProductController::searchAction`:
+        ```diff
+        - $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsForSearch(
+        + $paginationResult = $this->listedProductsFacade->getFilteredPaginatedForSearch(
+        ```
+    - in `ProductController::listByBrandAction`:
+        ```diff
+        - $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsForBrand(
+        -     $orderingModeId,
+        -     $page,
+        -     self::PRODUCTS_PER_PAGE,
+        -     $id
+        - );
+        + $paginationResult = $this->listedProductsFacade->getPaginatedForBrand(
+        +     $id,
+        +     $orderingModeId,
+        +     $page,
+        +     self::PRODUCTS_PER_PAGE
+        + );
+        ```
+    - in `CartController::addProductAjaxAxction`:
+        ```diff
+        - $accessories = $this->productAccessoryFacade->getTopOfferedAccessories(
+        -     $addProductResult->getCartItem()->getProduct(),
+        -     $this->domain->getId(),
+        -     $this->currentCustomer->getPricingGroup(),
+        -     self::AFTER_ADD_WINDOW_ACCESSORIES_LIMIT
+        );
+        + $accessories = $this->listedProductsFacade->getAccessories(
+        +     $addProductResult->getCartItem()->getProduct()->getId(),
+        +     self::AFTER_ADD_WINDOW_ACCESSORIES_LIMIT
+        + );
+        ```
+    - in `HomepageController::indexAction`:
+        ```diff
+        - $topProducts = $this->topProductFacade->getAllOfferedProducts(
+        -     $this->domain->getId(),
+        -     $this->currentCustomer->getPricingGroup()
+        - );
+        + $topProducts = $this->listedProductsFacade->getAllTop();
+        ```
+- edit your `productListMacro.html.twig` so it now works with instances of `ListedProductView` instead of `Product` entities
```

Comment:
I don't understand... what am I supposed to do? It seems I have to edit `productListMacro.html.twig`, please write down how.... the diff format is great!